License2Deploy/rolling_deploy.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/python
+
+import logging
+import boto.ec2 as ec2
+import boto.ec2.autoscale as a
+import boto.ec2.elb as e
+from sys import exit, argv
+from time import sleep, time
+
+class RollingDeploy(object):
+
+  def __init__(self, env=None, project=None, buildNum=None, ami_id=None, profile_name='default'):
+    self.env = env
+    self.project = project.replace('-','')
+    self.buildNum = buildNum
+    self.ami_id = ami_id
+    self.region = self.determine_region(self.env)
+    self.profile_name = profile_name
+    self.exit_error_code = 2
+
+  def aws_conn_auto(self):
+    return a.connect_to_region(self.region, profile_name=self.profile_name)
+
+  def aws_conn_ec2(self):
+    return ec2.connect_to_region(self.region, profile_name=self.profile_name)
+
+  def aws_conn_elb(self):
+    return e.connect_to_region(self.region, profile_name=self.profile_name)
+
+  def determine_region(self, env):
+    if env == 'qa':
+      return 'us-west-1'
+    elif env in ['stg', 'prd']:
+      return 'us-east-1'
+    else:
+      return env # This method is somewhat specific to our environment, so I am allowing the option for env to equal an actual region and will return the original value supplied
+
+  def get_ami_id_state(self, ami_id):
+    try:
+      ami_obj = self.aws_conn_ec2().get_all_images(image_ids=ami_id)
+    except Exception as e:
+      logging.error("Unable to get ami-id, please investigate: {0}".format(e))
+      exit(self.exit_error_code)
+    return ami_obj[0]
+
+  def wait_ami_availability(self, ami_id, timeout=5):
+    ''' Timeout should be in minutes '''
+    timeout = time() + 60*timeout
+    while True:
+      ami = self.get_ami_id_state(ami_id).state
+      if time() < timeout and ami == 'available':
+        logging.info("AMI {0} is ready".format(ami_id))
+        break
```

Comment:
Suggestion:
You could just return True here in place of the break. 
It makes the code a bit easier to read/determine where the action is happening.